File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
            
    int n  = nums.size();
    k = k % n;

    vector<int> temp(n);

    for(int i = 0 ; i < k ; i++){
        temp[i] = nums[n - k + i];
    }

     for (int i = k; i < n; i++) {
        temp[i] = nums[i - k];
    }
    nums = temp;

    for(int i = 0 ; i < n ; i++){
        cout<<nums[i];
    }

    }
    
};

File: 0189-rotate-array/0189-rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
  
    let n  = nums.length;
    k = k % n;

    let temp = new Array(0);

    for(let i = 0 ; i < k ; i++){
        temp[i] = nums[n - k + i];
    }

     for (let i = k; i < n; i++) {
        temp[i] = nums[i - k];
    }
    
      for (let i = 0; i < n; i++) {
        nums[i] = temp[i];
    }
};